### PR TITLE
add Controller object

### PR DESCRIPTION
**Pull request by loteque**
* Add controller object
* gun component uses controller object

### CODE REVIEW

**Comment by loteque** [comment]
* This technically closes #168
* ...but I think more work should be done for it to be useful if we want couch coop. We could close this now but I wouldn't recommend using it until we handle multiple player inputs, which will be done here and in a `PlayerManager`. 
  * If we use this now and add multiplayer later we will have to change code in every file we use this in.
* We are using two different methods of handling input currently, how we handle input in player controller is different from how we've had to handle input in gun controller and I think we should bring these into alignment, especially if we want to do coop.

File: project/src/guns/GunComponent.gd [hunk 1]
```diff
@@ -1,75 +1,54 @@
...
 	if aim_direction_vector != Vector2.ZERO:
 		var rotation_angle = aim_direction_vector.angle()
 		if is_facing_left:
-			_joypad_rotate_gun_left(rotation_angle)
+			self.rotation = joypad.get_right_stick_rotation(self, rotation_angle, 0.5)
 		else:
-			_joypad_rotate_gun_right(rotation_angle)
+			self.rotation = joypad.get_right_stick_inverse_rotation(self, rotation_angle, 0.5)
+
+func _init():
```
Comment (loteque) [inline project/src/guns/GunComponent.gd:44]: These should likely be called in player controller. `PlayerController` should use a controller object.

File: project/Controller.gd [hunk 2]
```diff
@@ -0,0 +1,32 @@
+extends Node
+class_name Controller
+
+enum {JOYPAD, MOUSE}
+var type: int
+var dev_id: int
```
Comment (loteque) [inline project/Controller.gd:6]: These are the vars that will need to be set on each controller object that gets instantiated in multiplayer.

**Comment by loteque** [comment]
* Controller.gd
	* Update to take responsibility away from PlayerController and GunComponent the control methods
* GunComponent.gd
	* remove methods that are handled by controller
* PlayerController.gd
	* use new Controller methods
	* remove responsibility to handle inputs on controllers

File: project/src/Controller.gd [hunk 3]
```diff
@@ -55,14 +57,14 @@ class Joypad extends Controller:
...
-		return get_stick_direction_vector(RSL, RSR, RSU, RSD).angle()
 	
-	## Returns new angle of stick rotation for aimed_node if the
-	## current controll vector not ZERO, else returns the current 
-	## angle of the node we are aiming.
+	## Returns new angle of stick rotation if the current
+	## controller vector not ZERO, else returns the previous value.
 	func get_aim_angle(_origin: Node2D) -> float:
-		return get_stick_direction_vector(RSL, RSR, RSU, RSD).angle()
+		var stick_vector = _get_stick_direction_vector(RSL, RSR, RSU, RSD)
```
Comment (jonathandavidlewis) [inline project/src/Controller.gd:66]: @loteque This is the fix for the bug you pointed out.

**Pull request merged**